### DISCUSSION TITLE
Installation on existing basedriver and gradle

### DISCUSSION THREAD

**Discussion by ganjarpanji** [AppiumTestDistribution/AppiumTestDistribution#818] [General]
Currently im using tech stack for my ui test as follows :
- appium
- no cucumber
- TestNg
- gradle
is  it possible to integrate ATD to test that have existing `BaseDriver` ? my test is not using cucumber, is the `testrunner` also needed as cukes? any sample of the integrated basedriver appium with ATD?
and how i would run my test using ATD from gradle? thanks in advance!

**Comment by saikrishna321** [comment]
What do we mean by basedriver? Can you please be more specific?

**Comment by ganjarpanji** [comment]
hi @saikrishna321 thanks for your response, basically i have a specific class that initiates the appium session before and we name it as `BaseDriver`

**Comment by saikrishna321** [comment]
@ganjarpanji 
1. ATD handles the driver session and you don't need to initiate any appium session at your end.
2.  If you have data set for each test you need to use @BeforeTest https://github.com/AppiumTestDistribution/PageObjectPatternAppium/blob/3e781f323437c25d14c218ef8b0550d56c2337f6/src/test/java/com/appium/tests/ScrollTest.java#L19 
3. If you need to have custom listener refer this - https://github.com/AppiumTestDistribution/AppiumTestDistribution/wiki/Applitool-Listener

**Comment by saikrishna321** [comment]
@ganjarpanji Does this help you
